### ISSUE TITLE
Docker image tags not updated correctly

### ISSUE THREAD

**Issue by taylorsilva** [concourse/concourse#5591]
## Summary

* `latest` tag on docker hub was not updated
* the `6.0` tag was updated at the same time as `6.1` was released. Other `6.0` tags like `6.0.0` were not updated.
```
$ docker pull concourse/concourse:6.0
$ docker run -it concourse/concourse:6.0 web -v
6.1.0
```

We should fix the `6.0` and `latest` tag and figure out why this happened.

https://hub.docker.com/r/concourse/concourse/tags

<img width="320" alt="Screen Shot 2020-05-12 at 5 47 23 PM" src="https://user-images.githubusercontent.com/4583837/81749074-f8d64600-9478-11ea-83d0-d59be02c45bd.png">

**Comment by cirocosta** [comment]
oooh yeah

https://ci.concourse-ci.org/teams/main/pipelines/release-6.1.x/jobs/publish-image/builds/1

![Screenshot from 2020-05-13 09-11-32](https://user-images.githubusercontent.com/3574444/81816798-d7696e80-94f9-11ea-879b-e860b429a50c.png)

ubuntu seems a bit weird too

![Screenshot from 2020-05-13 09-12-29](https://user-images.githubusercontent.com/3574444/81816851-e8b27b00-94f9-11ea-9ffb-f6c98064886e.png)

**Comment by vito** [comment]
So a few things went wrong here.

1. The pipeline configures `concourse-image` with `tag: ((latest_release))` and then does a `put` to it. The `latest_release` var was set to `6.0`, so that's how the wrong tag got updated. I think the intent here was to fix the upgrade/downgrade jobs, but we didn't realize that the image is also pushed to within the same pipeline.
1. The pipeline has never pushed to the `6` tag, even though it's supposed to. This didn't happen because [this check](https://github.com/concourse/ci/blob/b9f6647b72beef6f4f13425436461477715ae761/tasks/scripts/docker-semver-tags#L34) checks equality, when it should really check if the minor version is greater-than-or-equal-to, which is difficult to do in Bash.
1. For the same reason, the `latest` tag was also not bumped. [This check](https://github.com/concourse/ci/blob/b9f6647b72beef6f4f13425436461477715ae761/tasks/scripts/docker-semver-tags#L40) needs to be a 'greater-than-or-equal'.

Problems 2 and 3 also affected the `ubuntu` variant, leaving `6-ubuntu` and `ubuntu` un-pushed.

I've gone and fixed all the tags manually:

* re-pushed 6.0 to `6.0`
* published 6.1 to `latest` and `6`
* published 6.1 ubuntu to `ubuntu` and `6-ubuntu`

@cirocosta and I also did a deep dive into https://github.com/concourse/registry-image-resource/issues/151 and researched tagging practices  across a bunch of repositories. I think just the sheer difficulty involved in this and how easy it is to mess up, we should prioritize this sooner rather than later. It's pretty involved so I might draft an RFC at some point. 🤔
